pkg/solana/fees/block_history_test.go

```diff
@@ -11,86 +12,366 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
-	"go.uber.org/zap/zapcore"
 
 	"github.com/smartcontractkit/chainlink-common/pkg/logger"
 	"github.com/smartcontractkit/chainlink-common/pkg/utils"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/mathutil"
 	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
 
 	"github.com/smartcontractkit/chainlink-solana/pkg/solana/client"
 	clientmock "github.com/smartcontractkit/chainlink-solana/pkg/solana/client/mocks"
 	cfgmock "github.com/smartcontractkit/chainlink-solana/pkg/solana/config/mocks"
 )
 
-func TestBlockHistoryEstimator(t *testing.T) {
+func TestBlockHistoryEstimator_InvalidBlockHistorySize(t *testing.T) {
+	// Setup
+	invalidDepth := uint64(0) // Invalid value to trigger error
+	rw := clientmock.NewReaderWriter(t)
+	rwLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
+		return rw, nil
+	})
+	cfg := cfgmock.NewConfig(t)
+	cfg.On("BlockHistorySize").Return(invalidDepth)
+
+	// Initialize estimator and expect an error due to invalid BlockHistorySize
+	_, err := NewBlockHistoryEstimator(rwLoader, cfg, logger.Test(t))
+	require.Error(t, err, "Expected error for invalid block history size")
+	assert.Equal(t, "invalid block history depth: 0", err.Error(), "Unexpected error message for invalid block history size")
+}
+
+func TestBlockHistoryEstimator_LatestBlock(t *testing.T) {
+	// Helper variables for tests
 	min := uint64(10)
-	max := uint64(1000)
+	max := uint64(100_000)
+	defaultPrice := uint64(100)
+	depth := uint64(1) // 1 is LatestBlockEstimator
+	pollPeriod := 100 * time.Millisecond
+
+	// Grabbing last block of multiple_blocks file to use as latest block
+	testBlocks := readMultipleBlocksFromFile(t, "./multiple_blocks_data.json")
+	lastBlock := testBlocks[len(testBlocks)-1]
+	lastBlockFeeData, _ := ParseBlock(lastBlock)
+	lastBlockMedianPrice, _ := mathutil.Median(lastBlockFeeData.Prices...)
 
 	rw := clientmock.NewReaderWriter(t)
 	rwLoader := utils.NewLazyLoad(func() (client.ReaderWriter, error) {
 		return rw, nil
 	})
-	cfg := cfgmock.NewConfig(t)
-	cfg.On("ComputeUnitPriceDefault").Return(uint64(100))
-	cfg.On("ComputeUnitPriceMin").Return(min)
-	cfg.On("ComputeUnitPriceMax").Return(max)
-	cfg.On("BlockHistoryPollPeriod").Return(100 * time.Millisecond)
-	lgr, logs := logger.TestObserved(t, zapcore.DebugLevel)
-	ctx := tests.Context(t)
-
-	// file contains legacy + v0 transactions
-	testBlockData, err := os.ReadFile("./blockdata.json")
-	require.NoError(t, err)
-	blockRes := &rpc.GetBlockResult{}
-	require.NoError(t, json.Unmarshal(testBlockData, blockRes))
+	rw.On("GetLatestBlock", mock.Anything).Return(lastBlock, nil)
+
+	t.Run("Successful Estimation", func(t *testing.T) {
+		// Setup
+		cfg := cfgmock.NewConfig(t)
+		setupConfigMock(cfg, defaultPrice, min, max, pollPeriod, depth)
+		estimator := initializeEstimator(tests.Context(t), t, rwLoader, cfg, logger.Test(t))
```

Comment:
Sorry one more nit. Could you initialize a context at the top like `ctx := tests.Context(t)` and use that in all of the calls in these sub tests? Same for the multiple blocks tests?
```suggestion
		estimator := initializeEstimator(ctx, t, rwLoader, cfg, logger.Test(t))
```